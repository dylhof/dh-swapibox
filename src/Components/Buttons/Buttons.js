import React from 'react';

const Buttons = (props) => {
  const { makePeople, makePlanets, makeVehicles } = props
  return(
    <div className='button-div'>
      <button className='people-btn' onClick={makePeople}>People</button>
      <button className='planets-btn' onClick={makePlanets}>Planets</button>
      <button className='vehicles-btn' onClick={makeVehicles}>Vehicles</button>
      <button>Favorites</button>
    </div>
  )
}

export default Buttons;