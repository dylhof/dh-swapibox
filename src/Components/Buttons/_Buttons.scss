.button-div {
  width: 100%;
  height: 250px;
  display: flex;
  justify-content: center;
  position: fixed;
  bottom: 0px;
  align-items: center;
  // z-index: 2;
  background-color: $background;
}

button {
  height: 200px;
  width: 200px;
  border-radius: 50%;
  border: 10px solid $swFontColor;
  background-color: $background;
  margin: 20px;
  color: $swFontColor
}

// .people-btn {
//   background-image: url(sw_people.jpg)
// }