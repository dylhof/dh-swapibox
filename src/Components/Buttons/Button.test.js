import React from 'react';
import { shallow } from 'enzyme';
import Buttons from './Buttons';

const fetchPeopleMock = jest.fn();
const fetchPlanetsMock = jest.fn();
const fetchVehiclesMock = jest.fn();

describe('Buttons', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Buttons 
        fetchPeople={fetchPeopleMock}
        fetchPlanets={fetchPlanetsMock}
        fetchVehicles={fetchVehiclesMock}
      /> 
    )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call the fetch people method when people is clicked', () => {
    wrapper.find('.people-btn').simulate('click');
    expect(fetchPeopleMock).toBeCalled();
  })

  it('should call the fetch planets method when planets is clicked', () => {
    wrapper.find('.planets-btn').simulate('click');
    expect(fetchPlanetsMock).toBeCalled();
  })

  it('should call the fetch vehicles method when vehicles is clicked', () => {
    wrapper.find('.vehicles-btn').simulate('click');
    expect(fetchVehiclesMock).toBeCalled();
  })
})