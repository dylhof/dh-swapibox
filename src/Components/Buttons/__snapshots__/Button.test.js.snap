// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should match the snapshot 1`] = `
<div
  className="button-div"
>
  <button
    className="people-btn"
    onClick={[MockFunction]}
  >
    People
  </button>
  <button
    className="planets-btn"
    onClick={[MockFunction]}
  >
    Planets
  </button>
  <button
    className="vehicles-btn"
  >
    Vehicles
  </button>
  <button>
    Favorites
  </button>
</div>
`;
